### PR TITLE
Feature design blockly blocks swen900132021 ag 206

### PR DESCRIPTION
**Pull request by LIHANGT**
Add 6 blocks to customBlocks.js - less_than, greater_than, print_var, set_to, edge_weight, distance_from_to

### CODE REVIEW

File: src/Component/Blockly/customBlocks.js [hunk 1]
```diff
@@ -176,3 +176,133 @@ registerCustom({
...
+      { type: 'input_value', name: 'NODE_2' },
+    ],
+    inputsInline: true,
+    output: null,
+  },
+  style: 'loop_blocks',
+  generator: (block) => {
+    const srcNodeVar = Blockly.JavaScript.statementToCode(block, 'NODE_1') || 'node';
+    const destNodeVar = Blockly.JavaScript.statementToCode(block, 'NODE_2') || 'node';
+    return `${srcNodeVar}.getDistance(${destNodeVar});\n`;
```
Comment (Ev-anz) [inline src/Component/Blockly/customBlocks.js]: You need to implement this function to make this block work, currently there is no getDistance() function in a Node

File: src/Component/Blockly/customBlocks.js [hunk 2]
```diff
@@ -176,3 +176,133 @@ registerCustom({
...
+    args0: [
+      { type: 'input_value', name: 'EDGE' },
+    ],
+    inputsInline: true,
+    output: null,
+  },
+  style: 'loop_blocks',
+  generator: (block) => {
+    const edgeVar = Blockly.JavaScript.statementToCode(block, 'EDGE') || 'edge';
+    return `${edgeVar}.getWeight();\n`;
```
Comment (Ev-anz) [inline src/Component/Blockly/customBlocks.js:269]: Similar to below You need to implement this function to make this block work

File: src/Component/Blockly/customBlocks.js (hunk 1 above)
Comment (LIHANGT) [reply src/Component/Blockly/customBlocks.js]: Hi, I have added the function in bcakend. Could you please check the new PR in the API repo?

File: src/Component/Blockly/customBlocks.js (hunk 2 above)
Comment (LIHANGT) [reply src/Component/Blockly/customBlocks.js:269]: Also add the new function in the backend. Could you have a look?

Comment (maxplumley) [reply src/Component/Blockly/customBlocks.js:269]: this requires updates to ./src/graph

File: src/Component/Blockly/customBlocks.js (hunk 1 above)
Comment (maxplumley) [reply src/Component/Blockly/customBlocks.js]: this requires updates to ./src/graph

**Review by maxplumley — CHANGES_REQUESTED** [review]
UI graph code needs to be updated to support new node and edge functions

File: src/Component/Blockly/customBlocks.js (hunk 2 above)
Comment (LIHANGT) [reply src/Component/Blockly/customBlocks.js:269]: have added the functions in node.js and edge.js

File: src/Component/Blockly/customBlocks.js (hunk 1 above)
Comment (LIHANGT) [reply src/Component/Blockly/customBlocks.js]: have added the functions in node.js and edge.js

File: src/graph/node.js [hunk 3]
```diff
@@ -10,6 +10,17 @@ class Node {
     this.visited = false;
     this.graph = graph;
     this.focused = false;
+    this.distance = new Proxy({}, {
```
Comment (maxplumley) [inline src/graph/node.js]: any reason why this cant just be an object ie {}?

Comment (LIHANGT) [reply src/graph/node.js]: I explained the reason on the PR of backend repo

**Review by maxplumley — APPROVED** [review]

**Pull request merged**